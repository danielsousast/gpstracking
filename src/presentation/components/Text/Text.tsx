/* eslint-disable react-native/no-inline-styles */
import React, {PropsWithChildren} from 'react';
import styled from 'styled-components/native';

interface Props extends PropsWithChildren {
  preset?: 'title' | 'subtitle' | 'body' | 'label';
  mt?: number;
  color?: string;
  fontStyle?: 'normal' | 'italic';
}

const fontSizes = {
  title: 24,
  subtitle: 20,
  body: 16,
  label: 12,
};

export function Text({
  preset,
  children,
  mt = 0,
  color = '#000',
  fontStyle = 'normal',
}: Props) {
  return (
    <TextRN
      style={{
        fontSize: fontSizes[preset || 'body'],
        fontWeight: preset === 'title' ? 'bold' : 'normal',
        fontStyle,
        marginTop: mt,
        color,
      }}>
      {children}
    </TextRN>
  );
}

export const TextRN = styled.Text``;
