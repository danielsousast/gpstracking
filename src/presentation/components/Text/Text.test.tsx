import {Text} from './Text';
import {render} from '@testing-library/react-native';

describe('Text Component', () => {
  test('should render', () => {
    const rendered = render(<Text>Test</Text>);
    expect(rendered).toBeTruthy();
  });

  test('should render with default props', () => {
    const rendered = render(<Text>Test</Text>);
    expect(rendered).toMatchSnapshot();
  });

  test('should render with preset', () => {
    const rendered = render(<Text preset="title">Test</Text>);
    expect(rendered).toMatchSnapshot();
  });
});
