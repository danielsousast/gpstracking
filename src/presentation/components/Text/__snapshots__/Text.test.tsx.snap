// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text Component should render with default props 1`] = `
<Text
  style={
    [
      {},
      {
        "color": "#000",
        "fontSize": 16,
        "fontStyle": "normal",
        "fontWeight": "normal",
        "marginTop": 0,
      },
    ]
  }
>
  Test
</Text>
`;

exports[`Text Component should render with preset 1`] = `
<Text
  style={
    [
      {},
      {
        "color": "#000",
        "fontSize": 24,
        "fontStyle": "normal",
        "fontWeight": "bold",
        "marginTop": 0,
      },
    ]
  }
>
  Test
</Text>
`;
